app/src/main/java/com/example/readnews/headlines/TopHeadlinesFragment.kt

```diff
@@ -101,5 +138,25 @@ class TopHeadlinesFragment : Fragment() {
             }
         }
     }
+
+    var callback: HeadlineListener? = null
```

Comment:
ça fonctionne, mais ce n'est pas vraiment la solution attendue.. 
Les callbacks sont à éviter au maximum.

Dans ton cas l'erreur se matérialise par le fait que le réseau soit désactivé, mais si ton ws de news te renvoie une erreur (ex: code 500), tu dois quand même la gérer..

je te conseille de regarder https://medium.com/@douglas.iacovelli/how-to-handle-errors-with-retrofit-and-coroutines-33e7492a912
(c'est un exemple)